### PR TITLE
support loading a subset of tf graph and support control dependency in graph

### PR DESCRIPTION
**Pull request by yangw1234**
## What changes were proposed in this pull request?

1. support loading a subset of tf graph
2. Change `Context` key type to String for future development convenience.
3. handle tensorflow control dependency

## How was this patch tested?

add unit tests

### CODE REVIEW

File: spark/dl/src/main/scala/com/intel/analytics/bigdl/utils/tf/TensorflowLoader.scala [hunk 1]
```diff
@@ -105,29 +127,75 @@ object TensorflowLoader{
...
-      name2Node.valuesIterator.filter(_.nextNodes.length == 0).toArray
+      name2Node.valuesIterator.filter(_.nextNodes.isEmpty).toArray
     } else {
       val results = name2Node.valuesIterator.toArray.filter(n =>
         outputNodeNames.contains(n.element.getName))
       require(results.length == outputNodeNames.length, "Invalid outputNode names")
       results
     }
 
+    def connect(nodes: Seq[Node[NodeDef]]): Seq[String] = {
```
Comment (yiheng) [inline spark/dl/src/main/scala/com/intel/analytics/bigdl/utils/tf/TensorflowLoader.scala]: private

File: spark/dl/src/main/scala/com/intel/analytics/bigdl/utils/tf/TensorflowLoader.scala [hunk 2]
```diff
@@ -84,13 +87,32 @@ object TensorflowLoader{
   }
 
   /**
+   * Parse a tensorflow model protobuf file, read a list of op nodes from it
+   * @param graphProtoTxt where is the tf protobuf file
+   * @return
+   */
+  private[bigdl] def parseTxt(graphProtoTxt: String) : List[NodeDef] = {
```
Comment (yiheng) [inline spark/dl/src/main/scala/com/intel/analytics/bigdl/utils/tf/TensorflowLoader.scala]: what's the difference from parse

Comment (yangw1234) [reply spark/dl/src/main/scala/com/intel/analytics/bigdl/utils/tf/TensorflowLoader.scala]: parseTxt process model.pbtxt, parse process model.pb

**Comment by yangw1234** [comment]
pr validation pass

**Pull request merged**